### PR TITLE
challenge_exit

### PR DESCRIPTION
**Pull request by parthgargava**

### CODE REVIEW

File: contracts/RootChain.sol [hunk 1]
```diff
@@ -111,6 +111,19 @@ contract RootChain {
         uint256 _spendingTxSignatures,
         uint256 _spendingTxConfirmationSignatures
     ) public {
+		uint256 eUtxoPos = _encodedSpendingTx.getUtxoPos(_exitingUtxoPosition);
+
+        uint256 _txindex  = (_spendingTxPosition % 1000000000) / 10000;
+        bytes32 root = plasmaChain[_spendingTxPosition / 1000000000].root; //need to make plasma chain/child chain
+        var txHash = keccak256(_encodedSpendingTx);
```
Comment (smartcontracts) [inline contracts/RootChain.sol:118]: Let's replace `var` with the corresponding type (e.x. `bytes txHash`)

File: contracts/RootChain.sol [hunk 2]
```diff
@@ -111,6 +111,19 @@ contract RootChain {
...
+        var confirmationHash = keccak256(txHash, root);
+        var merkleHash = keccak256(txHash, _spendingTxSignatures);
```
Comment (smartcontracts) [inline contracts/RootChain.sol:120]: Let's get rid of `_spendingTxSignatures` here entirely and instead use `keccak256(txHash)`.

File: contracts/RootChain.sol [hunk 3]
```diff
@@ -111,6 +111,19 @@ contract RootChain {
         uint256 _spendingTxSignatures,
         uint256 _spendingTxConfirmationSignatures
     ) public {
+		uint256 eUtxoPos = _encodedSpendingTx.getUtxoPos(_exitingUtxoPosition);
+
+        uint256 _txindex  = (_spendingTxPosition % 1000000000) / 10000;
```
Comment (smartcontracts) [inline contracts/RootChain.sol:116]: We can make use of `PlasmaCore` here instead:
`uint256 _txindex = getTxIndex(_spendingTxPosition)`

Likewise for `root`.

File: contracts/RootChain.sol [hunk 4]
```diff
@@ -111,6 +111,19 @@ contract RootChain {
...
+        bytes32 root = plasmaChain[_spendingTxPosition / 1000000000].root; //need to make plasma chain/child chain
```
Comment (smartcontracts) [inline contracts/RootChain.sol:117]: I think it'd be cleaner to separate the `blknum` variable here instead of doing it in one line.

**Review by smartcontracts — CHANGES_REQUESTED** [review]
Requesting a few small changes.

**Review by parthgargava — COMMENTED** [review]
Ok, the changes make sense

**Pull request closed without merging**